mesh/mesh.go

```diff
@@ -276,18 +342,34 @@ func (m *Mesh) GetLatestView() []types.BlockID {
 
 func (m *Mesh) AddBlock(blk *types.Block) error {
 	m.Debug("add block %d", blk.ID())
+	m.AtxDB.ProcessBlockATXs(blk) // change this to return error if process failed
```

Comment:
Handle comment? There is an important logic behind validating ATXs, you should handle it as part of this PR or the next one (syncing ATXs)

Reply:
that logic is in sync and in the block listener, im not sure it should be here but i definitely agree that we need some validator that contains this logic